### DISCUSSION TITLE
"Home Z and Disable" doesn't disable steppers when Z-Offset not equal to Zero

### DISCUSSION THREAD

**Discussion by Ergonomicmike** [mriscoc/Ender3V2S1#988] [Q&A]
I was readjusting my Z-height micro-switch this evening, and was using the "Home Z and Disable (stepper motors)" function for the first time while making the adjustment.

Although Z would home, the steppers didn't disable.

Then I realized that I had Z-Offset still set to what I was using before making the adjustment on the switch.

After I set Z-Offset to Zero, then the steppers disabled.

I don't know if the former behavior is the desired behavior. On one hand, I can see that, after Homing Z, the Z-motor would have to stay engaged in order to move the gantry to the Z-Offset. On the other hand, I expected that the steppers would disable after that.

(This is an Ender 3 v2, Manual Mesh (no Probe).)

**Comment by mriscoc** [comment]
Sorry, but I couldn't replicate your problem.
Tested with Ender3V2 and 4.2.2 board, MM version.

**Comment by Ergonomicmike** [comment]
Gracias por comprobarlo.

I tested it again tonight from a cold start and I could not replicate my problem either.

(One difference from when I posted is that I have recompiled the updated version of your software.)

IIRC, it was after a cold start too when I was readjusting the Z-height micro-switch. 

I will continue to test for this during day-to-day use to see if there is some combination to trigger it.

But for now, no importa.
